### DISCUSSION TITLE
下蹲使用盾牌

### DISCUSSION THREAD

**Discussion by WhiteElephant-abc** [TouchController/TouchController#120] [输入控制逻辑]
建议添加下蹲使用盾牌的功能，现有逻辑会出现和 #115 同样的问题

**Comment by fifth-light** [comment]
问题在于 Java 版的副手使用场景还很多，除了盾牌以外还有很多场景，要不一起在这里讨论一下怎么做

**Reply by WhiteElephant-abc** [reply]
问一下java主副手的原版逻辑是什么

**Reply by fengzhou0w0** [reply]
应该是左键使用主手，挖掘攻击等，右键优先级：交互＞使用副手功能＞主手放置

**Reply by WhiteElephant-abc** [reply]
啊？副手右键优先级比左手高吗

**Reply by fifth-light** [reply]
这个分两种情况：

攻击：只会使用主手攻击。

使用：先会尝试使用主手物品，如果使用失败则会使用副手。

**Reply by WhiteElephant-abc** [reply]
对了，现有逻辑还非常容易误触，就是拖屏幕的时候很容易就把盾牌举起来，然后就走不动路了，挺难受的

**Reply by fifth-light** [reply]
> 拖屏幕的时候很容易就把盾牌举起来

我记得我在设置里有个阈值，可以设置手指移动多少算长按，可以调节这个值试试。

**Comment by WhiteElephant-abc** [comment]
但是TC更复杂，因为点击屏幕同时对应了左键和右键
